app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -43,6 +44,40 @@ def destroy
       respond_for_destroy(:resource => @host)
     end
 
+    api :POST, "/hosts/subscriptions/", N_("Register a host with subscription and information.")
+    param :name, String, :desc => N_("Name of the host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host")
+    param :facts, Hash, :desc => N_("Key-value hash of subscription-manager facts") do
+      param :fact, String, :desc => N_("Any number of sbscription facts about this host")
+    end
+    param :hypervisor_guest_uuids, Array, :desc => N_("UUIDs of the virtual guests from the host's hypervisor")
+    param :installed_products, Array, :desc => N_("List of products installed on the host")
```

Comment:
It needs to be in apipie for use by consumers of the api (hammer).

There's room for doc on this, of course, but is it going to block merge? The params existed previously.

The products are not always present as katello products. These are based upon, at the moment, Red Hat shipped product SKUs.